src/app/home/employee/components/create-employee/create-employee.component.ts

```diff
@@ -1,206 +1,227 @@
-        ],
-        sex: ['', [Validators.required]],
-        birthDay: ['', [Validators.required]],
-        currentResidence: ['', [Validators.maxLength(255), emojiValidator]],
-        address: ['', [Validators.maxLength(255), emojiValidator]],
-        joinDate: [''],
-        hireDate: '',
-        avt: [''],
-      }),
-      contactInfo: this.fb.group({
-        email: [
-          '',
-          [Validators.required, Validators.maxLength(50), emojiValidator],
-        ],
-        phone: ['', [Validators.required, Validators.maxLength(11)]],
-        skypeId: [
-          '',
-          [Validators.required, Validators.maxLength(255), emojiValidator],
-        ],
-      }),
-      workingProcess: this.fb.group({
-        unit: [''],
-        position: [''],
-        workingTime: [''],
-        workingForm: [''],
-      }),
-      otherInfo: this.fb.group({
-        description: ['', [Validators.maxLength(500)]],
-        unit: ['', Validators.required],
-        position: ['', Validators.required],
-        cv: [''],
-        status: ['', [Validators.required]],
-      }),
-    });
-  }
-}
+import { Location } from '@angular/common';
+import { Component } from '@angular/core';
+import {
+  AbstractControl,
+  FormArray,
+  FormBuilder,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+import { CommonService } from 'src/app/core/services/common.service';
+import { emojiValidator } from 'src/app/core/services/helper/validator.service';
+import { EmployeeService } from 'src/app/core/services/http/employee.service';
+import { UnitTreeService } from 'src/app/core/services/state/uint-tree.service';
+import { LoadingService } from 'src/app/core/services/state/loading.service';
+import { ToastService } from 'src/app/core/services/helper/toast.service';
+import { toast } from 'src/app/shared/toastMessage';
+
+@Component({
+  selector: 'app-create-employee',
+  templateUrl: './create-employee.component.html',
+  styleUrls: ['./create-employee.component.scss'],
+})
+export class CreateEmployeeComponent {
+  public assetList = [];
+  public formData: FormData = new FormData();
+  constructor(
+    private location: Location,
+    private fb: FormBuilder,
+    private commonService: CommonService,
+    private unitTreeService: UnitTreeService,
+    private employeeService: EmployeeService,
+    private loadingService:LoadingService,
+    private toastService:ToastService
+  ) {}
+  public employeeForm: FormGroup;
+
+  handleBack(): void {
+    this.location.back();
+  }
+
+  getControl(
+    type: string,
+    control: string
+  ): AbstractControl | null | undefined {
+    return this.employeeForm.get(type)?.get(control);
+  }
+
+  onSubmit(): void {
+    this.commonService.markAsDirty(this.employeeForm);
+    if (this.employeeForm.valid) {
+      this.loadingService.setloading(true);
+      const data = {
+        employee_code: this.getControl('basicInfo', 'code')?.value,

```

Comment:
mấy chỗ này thử sử dụng get form nested thế này xem được không .
this.employeeForm.get(['basicInfo','code']).value,
đâu cần phải viết common cái này